edition/voxel_tool_terrain.cpp

```diff
@@ -37,9 +37,15 @@ Ref<VoxelRaycastResult> VoxelToolTerrain::raycast(Vector3 pos, Vector3 dir, floa
 				return false;
 
 			const Voxel &voxel = lib.get_voxel_const(v0);
+			if (voxel.get_geometry_type() == voxel.GEOMETRY_NONE)
+				return false;
+
 			if (voxel.is_transparent() == false)
 				return true;
 
+			if (voxel.is_transparent() && voxel.get_collision_aabbs().empty() == false)
```

Comment:
Technically, if the voxel has collision AABBs, an optional narrow phase should be run on them with a for loop I believe, because as in minecraft, a voxel model is not necessarily a cube. Yet water has a collision AABB, just not taken into account so there will still be room for refinements in raycast capabilities.
Your code should be enough for now, but it's going to change in the future anyways.